lib/flake.nix

```diff
@@ -41,18 +41,19 @@
       {
         inherit callLibs;
 
-        attrs = callLibs ./attrs.nix;
-        os = callLibs ./devos;
-        lists = callLibs ./lists.nix;
-        strings = callLibs ./strings.nix;
+        attrs = import ./attrs.nix { lib = prev; };
+        lists = import ./lists.nix { lib = prev; };
+        strings = import ./strings.nix { lib = prev; };
+
+        os = import ./devos { lib = final; };
 
         mkFlake = {
           __functor = callLibs ./mkFlake;
           evalArgs = callLibs ./mkFlake/evalArgs.nix;
           evalOldArgs = callLibs ./mkFlake/evalOldArgs.nix;
         };
 
-        pkgs-lib = callLibs ./pkgs-lib;
+        pkgs-lib = import ./pkgs-lib {lib = final; inherit nixpkgs deploy devshell};
```

Comment:
I guess it would be nice to pull up instantiation of nixpkgs with overlays up into the top level and work with a more familiar `pkgs` instead. ... another time, though.